CRM/Upgrade/Incremental/php/FourSeven.php

```diff
@@ -1232,6 +1233,37 @@ public static function pickActivityRevisionPolicy(CRM_Queue_TaskContext $ctx) {
   }
 
   /**
+   * Remove the contribution logging reports which have been broken for a very long time.
+   *
+   * @param \CRM_Queue_TaskContext $ctx
+   *
+   * @return bool
+   */
+  public static function removeContributionLoggingReports(CRM_Queue_TaskContext $ctx) {
+    if (class_exists('CRM_Report_Form_Contribute_LoggingDetail') || class_exists('CRM_Report_Form_Contribute_LoggingSummary')) {
+      // Perhaps the site has overridden these classes. The core ones are broken but they
+      // may have functional ones.
+      return TRUE;
+    }
+    $options = civicrm_api3('OptionValue', 'get', array('option_group_id' => 'report_template', 'options' => array('limit' => 0)));
```

Comment:
isn't this getting all reports? shouldn't we limit them to just the 2 that we care about?

Reply:
yeah, getting all & iterating to filter. I guess I could use IN - but I tested this one!

Reply:
Makes sense i guess